server/src/controllers/eventController.ts

```diff
@@ -143,6 +149,90 @@ eventController.post('/checkout', async (req: Request, res: Response) => {
   }
 });
 
+/**
+ * @swagger
+ * /2/events/image-upload:
+ *   post:
+ *     summary: Upload image to google cloud storage, make public, return link
+ *     tags:
+ *     - New Event API
+  *     requestBody:
+ *       description: Image File
+ *     responses:
+ *       200:
+ *         description: Image successfully uploaded.
+ *         content:
+ *           application/json:
+ *             schema: {url: string}
+ *       400:
+ *         description: bad request
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 error:
+ *                   type: string
+ *                   description: Error message from the server.
+ *       500:
+ *         description: Internal Server Error. An error occurred while processing the request.
+ */
+
+eventController.post('/image-upload', upload.single('file'), async (req: Request, res: Response) => {
+  const gcloudKey = process.env.GCLOUD_KEY;
+  const bucketName = process.env.GCLOUD_BUCKET;
+
+  if (gcloudKey === undefined || gcloudKey === '') {
+    console.log('Bucket key undefined!');
+    return res.status(500).send('Bucket key undefined!');
+  }
+  if (bucketName === undefined || bucketName === '') {
+    console.log('Bucket name undefined!');
+    return res.status(500).send('Bucket name undefined!');
+  }
+
+  const credentials = JSON.parse(gcloudKey);
+
+  const storage = new Storage({credentials: credentials});
+  const imgBucket = storage.bucket(`${bucketName}`);
+
+  if (!req.file) {
+    return res.status(400).send('No file passed to request!');
+  }
+
+  try {
+    validateWithRegex(
+      req.file.mimetype,
+      'Invalid input, not a valid image filetype!', 
+      new RegExp('^(image\/(jpe?g|png))')
```

Comment:
Does this regex also work for the capitalized versions of the extensions.

Reply:
Filetype is now converted to lowercase before being regex'd